.github/workflows/python-conda-test.yml

```diff
@@ -77,6 +84,7 @@ jobs:
     name: "Python ${{ inputs.python-version }}: conda"
     runs-on: ubuntu-20.04
     continue-on-error: ${{ inputs.experimental }}
+    permissions: read-all
```

Comment:
> I set workflow-wide GITHUB_TOKEN permissions to read-all (read-only).
the job doesn't write to github at all, so this shouldn't change anything
we also don't use the token in this workflow anyway

Does this mean that you can get the token from inside the tmate session, but it could only be used for reading info from the github API? e.g. you could use it to check commits or repo metadata, but not to, say, make PRs?

Reply:
So as far as I can tell, GITHUB_TOKEN needs to be accessed and inserted into the step ([docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input)).  As in we need to set it to an env var in the step.  Since we don't do that in the tmate step, we the tmate session doesn't have access.

So I think this line might actually be unnecessary, but I put it there as a precaution.  The [default permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) have some read-write scattered in there.